### PR TITLE
OSSM-3462 migrate maistra istio push image jobs

### PR DESCRIPTION
**Pull request by yxun**
This PR migrates maistra istio 2.3 and 2.4 branch push image jobs. And it migrates the maistra istio 2.3 integ tests as well. When all istio integ tests pass, a push image job will be triggered and push the PR number tag to quay.

### CODE REVIEW

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-maistra-istio-integ-aws-2-4

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-maistra-istio-integ-aws-2-4

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-maistra-istio-integ-aws-2-4

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.3-maistra-istio-integ-aws-2-3

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-maistra-istio-push-image-2-4

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-maistra-istio-push-image-2-4

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.3-maistra-istio-push-image-2-3

**Review by bartoszmajsak — APPROVED** [review]
I wonder if we could have a clean-up job that deletes the image from the repository on a PR close. This could be easily done by using e.g. `skopeo delete` (can also be done through regular API calls by using `curl` if we don't want to add yet another tool to the toolbelt). I'm not saying it's part of this particular PR but might be a good addition to the process.

File: ci-operator/jobs/maistra/istio/maistra-istio-maistra-2.4-presubmits.yaml [hunk 1]
```diff
@@ -213,6 +213,69 @@ presubmits:
...
+      - name: hive-hive-credentials
+        secret:
+          secretName: hive-hive-credentials
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )maistra-istio-push-image-2-4,?($|\s.*)
```
Comment (bartoszmajsak) [inline ci-operator/jobs/maistra/istio/maistra-istio-maistra-2.4-presubmits.yaml:278]: I wonder if we can't simplify the triggers to  be more action-oriented. It's not really `/test action` but `/action`. So maybe `/push-image` would be easier to guess :) WDYT @yxun

Comment (yxun) [reply ci-operator/jobs/maistra/istio/maistra-istio-maistra-2.4-presubmits.yaml:278]: @bartoszmajsak  Thanks for your review. The files under ci-operator/jobs/ are auto generated from openshift-ci based on the files under ci-operator/config/ 
We don't have control of jobs/ files. They will be overwritten automatically in one of the presubmit job in each openshift CI PR. 

I can check more if we can customize their trigger config in a prow plugin setting later.

**Pull request merged**